classes/Category.php

```diff
@@ -2054,6 +2054,23 @@ public function getProductsWs()
 		ORDER BY `position` ASC');
     }
 
+    /**
+     * Get products of the category
+     * The returned list of products is used during category deletion to re-assign them to main category
+     *
+     * @return array|false|mysqli_result|PDOStatement|resource|null
+     */
+    public function getCategoryProducts()
+    {
+        return Db::getInstance()->executeS('
+        SELECT p.`id_product`
+        FROM `' . _DB_PREFIX_ . 'product` p
+        ' . Shop::addSqlAssociation('product', 'p') . '
+        LEFT JOIN `' . _DB_PREFIX_ . 'category_product` cp
+        ON cp.`id_product` = p.`id_product`
+        WHERE cp.`id_category` = ' . (int) $this->id);
```

Comment:
Better : could you `sprintf` for the query ?

Reply:
Fixed. (But I feel I should make it better by using category Repository and rely on ORM, maybe inject the service `category.repository` somewhere :thinking: .)